#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kuka/youbot/protos/Youbot.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.39506129833983905 -0.7017242294721342 -0.5928825147748752 4.53916476017376
  position -24.085028482726774 -6.373722344318176 19.963164521409503
  follow "youBot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  rotation 0 0 -1 0.406
  floorSize 60 60
  floorTileSize 60 60
  floorAppearance PBRAppearance {
    baseColorMap ImageTexture {
      url [
        "../../Maze.png"
      ]
    }
    roughness 1
    metalness 0
  }
  wallThickness 2
  wallHeight 2
}
Youbot {
  hidden position_0_0 859.849520964695
  hidden position_0_1 3691.0727146349
  hidden position_0_2 1210.678014788609
  hidden position_0_3 3341.019261273687
  hidden position_17_0 -8.875620429107287e-06
  hidden rotation_18 0 0 -1 8.87563997328823e-06
  hidden position_18_0 -2.081111092736875e-07
  hidden rotation_19 0 1 0 2.086162567138673e-07
  hidden position_19_0 1.7582388113614408e-06
  hidden rotation_20 0 -1 0 1.75833702087408e-06
  hidden position_20_0 -4.770474811831689e-08
  hidden rotation_21 0 1 0 5.16191365590357e-08
  hidden position_21_0 -1.6962196947615092e-08
  hidden position_22_0 1.364285483145358e-05
  hidden position_22_1 1.3549922765656765e-05
  hidden translation_23 0 1.364285483145358e-05 0.09
  hidden translation_24 0 -1.3549922765656765e-05 0.09
  hidden rotation_25 0 1 0 5.336319188271228
  hidden rotation_26 0 -0.9999999999999999 0 3.4402459866968815
  hidden rotation_27 0 1 0 4.306435810128458
  hidden rotation_28 0 -1 0 1.6353221458531262
  translation -24.1162 -1.9914 0.107852
  rotation -0.0028221288613435765 0.0018648143110178747 0.9999942790117728 1.2051941098954368
  controller "youbot_controller"
  bodySlot [
    DistanceSensor {
      translation 0.308319 0.000267428 -0.03
      rotation 0.9999999999971059 -1.620879999995309e-06 -1.7779899999948542e-06 -1.5707953071795862
      children [
        Solid {
          translation 0.01 -0.002 0.0006
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.115
              }
            }
          ]
        }
      ]
      name "w_front"
      lookupTable [
        0 0 0
        1.6 1000 0
      ]
      type "infra-red"
      numberOfRays 3
    }
    DistanceSensor {
      translation -0.291681 0.000267394 -0.0299985
      rotation -1.2606205012942948e-06 0.7071072811860076 -0.70710628118561 3.14159
      children [
        Solid {
          translation 0.01 -0.002 0.0006
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.115
              }
            }
          ]
        }
      ]
      name "w_back"
      lookupTable [
        0 0 0
        0.8 1000 0
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.028319 0.180267 -0.03
      rotation 0.5773492691885823 0.5773502691890485 -0.5773512691895147 -2.094395307179586
      children [
        Solid {
          translation 0.01 -0.002 0.0006
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.115
              }
            }
          ]
        }
      ]
      name "w_left"
      lookupTable [
        0 0 0
        0.9 1000 0
      ]
      type "infra-red"
    }
    DistanceSensor {
      translation 0.028319 -0.199733 -0.03
      rotation -0.5773512691895147 0.5773502691890485 -0.5773492691885823 2.0944
      children [
        Solid {
          translation 0.01 -0.002 0.0006
          rotation 0.5773509358554485 0.5773489358556708 0.5773509358554485 2.0944
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.115
              }
            }
          ]
        }
      ]
      name "w_right"
      lookupTable [
        0 0 0
        0.9 1000 0
      ]
      type "infra-red"
    }
    Camera {
      translation 0.28 0 0.04
      children [
        Solid {
          rotation 0 1 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Cylinder {
                height 0.1
                radius 0.02
              }
            }
          ]
        }
      ]
      name "cam"
      fieldOfView 1
      width 640
      height 640
      far 10
      recognition Recognition {
        frameThickness 5
      }
    }
    DistanceSensor {
      translation 0.312236 -0.101023 -0.04
      rotation 0.1294279597735375 -0.9831056944488314 -0.1294279597735375 -1.5878353071795859
      children [
        DEF ls_right Solid {
          translation 0 0.0496411 -0.028
          rotation 1 0 0 1.0472
          children [
            DEF ls_right Shape {
              appearance PBRAppearance {
                baseColor 1 0.4 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.02 0.02 0.115
              }
            }
          ]
          name "ls_right"
        }
      ]
      name "ls_right"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.320471 0.101274 -0.0419506
      rotation 0.6232131723564657 -0.5529947563641509 -0.552993798550349 -2.0944003061004253
      children [
        DEF ls_left Solid {
          translation 0 0.0496411 -0.028
          rotation 1 0 0 1.0472
          children [
            DEF ls_left Shape {
              appearance PBRAppearance {
                baseColor 1 0.4 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.02 0.02 0.115
              }
            }
          ]
          name "ls_left"
        }
      ]
      name "ls_left"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.338319 0.000267444 -0.04
      rotation -0.44721312362157567 0.7745972141192264 0.44721312362157567 1.82348
      children [
        DEF ls_middle Solid {
          translation 0 0.0496411 -0.028
          rotation 1 0 0 1.0472
          children [
            DEF ls_middle Shape {
              appearance PBRAppearance {
                baseColor 1 0.4 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.02 0.02 0.115
              }
            }
          ]
          name "ls_middle"
        }
      ]
      name "ls_middle"
      type "infra-red"
    }
  ]
}
Solid {
  translation -21.9124 6.37087 0.0434087
  rotation -1.1633899997478673e-05 1.7265499996258177e-05 0.9999999997832776 -1.962365307179586
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 14.15 0.6 0.6
      }
    }
  ]
  name "R1"
  boundingObject USE b1
}
Solid {
  translation -9.60145 14.2574 0.0436979
  rotation 1.1633938246886936e-05 -1.7265510063038425e-05 -0.9999999997832768 1.96236506221651
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 11.16 0.6 0.6
      }
    }
  ]
  name "R1(34)"
  boundingObject USE b1
}
Solid {
  translation 2.51892 20.3453 0.0540049
  rotation 1.1633938246886936e-05 -1.7265510063038425e-05 -0.9999999997832768 1.96236506221651
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 10.7 0.6 0.6
      }
    }
  ]
  name "R1(2)"
  boundingObject USE b1
}
Solid {
  translation 10.9442 16.9322 0.0543195
  rotation 1.1633938246886936e-05 -1.7265510063038425e-05 -0.9999999997832768 1.96236506221651
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 10.7 0.6 0.6
      }
    }
  ]
  name "R1(3)"
  boundingObject USE b1
}
Solid {
  translation 8.43259 17.19 0.0542354
  rotation 1.1633938246886936e-05 -1.7265510063038425e-05 -0.9999999997832768 1.96236506221651
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 9.4 0.6 0.6
      }
    }
  ]
  name "R1(4)"
  boundingObject USE b1
}
Solid {
  translation 4.60526 18.7149 0.0540932
  rotation 1.1633938246886936e-05 -1.7265510063038425e-05 -0.9999999997832768 1.96236506221651
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 9.3 0.6 0.6
      }
    }
  ]
  name "R1(5)"
  boundingObject USE b1
}
Solid {
  translation 7.52963 23.0345 0.0541303
  rotation -1.5583199997656748e-05 7.609519998855754e-06 0.9999999998496296 -3.009565307179586
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 2.8 0.6 0.6
      }
    }
  ]
  name "R1(7)"
  boundingObject USE b1
}
Solid {
  translation 7.14363 25.4849 0.0540866
  rotation -1.5583199997656748e-05 7.609519998855754e-06 0.9999999998496296 -3.009565307179586
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 5.7 0.6 0.6
      }
    }
  ]
  name "R1(9)"
  boundingObject USE b1
}
Solid {
  translation 11.3627 23.772 0.0542443
  rotation 2.7138799979181864e-06 -3.9074699970025856e-05 -0.9999999992329014 0.91517
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 5.5 0.6 0.6
      }
    }
  ]
  name "R1(10)"
  boundingObject USE b1
}
Solid {
  translation 9.45867 22.2981 0.0542014
  rotation -1.9242099996420544e-05 -1.3364399997513927e-06 0.9999999998139779 2.22642
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 2.6 0.6 0.6
      }
    }
  ]
  name "R1(8)"
  boundingObject USE b1
}
Solid {
  translation -8.21109 11.9651 0.0437723
  rotation 1.1633938246886936e-05 -1.7265510063038425e-05 -0.9999999997832768 1.96236506221651
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 8.7 0.6 0.6
      }
    }
  ]
  name "R1(35)"
  boundingObject USE b1
}
Solid {
  translation -3.90237 10.9759 0.0439237
  rotation 1.1633938246886936e-05 -1.7265510063038425e-05 -0.9999999997832768 1.96236506221651
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 7.59 0.6 0.6
      }
    }
  ]
  name "R1(36)"
  boundingObject USE b1
}
Solid {
  translation -2.0603 9.62049 0.0440009
  rotation 1.1633938246886936e-05 -1.7265510063038425e-05 -0.9999999997832768 1.96236506221651
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 8.7 0.6 0.6
      }
    }
  ]
  name "R1(42)"
  boundingObject USE b1
}
Solid {
  translation -13.2072 5.21051 0.0485584
  rotation 1.1633938246886936e-05 -1.7265510063038425e-05 -0.9999999997832768 1.96236506221651
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 4.26 0.6 0.6
      }
    }
  ]
  name "R1(24)"
  boundingObject USE b1
}
Solid {
  translation -15.8939 -1.2744 0.0485584
  rotation 1.1633938246886936e-05 -1.7265510063038425e-05 -0.9999999997832768 1.96236506221651
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 5.19 0.6 0.6
      }
    }
  ]
  name "R1(28)"
  boundingObject USE b1
}
Solid {
  translation -17.9309 -1.12553 0.0484911
  rotation 1.1633938246886936e-05 -1.7265510063038425e-05 -0.9999999997832768 1.96236506221651
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 4.38 0.6 0.6
      }
    }
  ]
  name "R1(29)"
  boundingObject USE b1
}
Solid {
  translation -16.3386 9.0942 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 3.7 0.6
      }
    }
  ]
  name "R1(6)"
  boundingObject USE b1
}
Solid {
  translation -4.55263 15.1401 0.0387247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 4.94 0.6
      }
    }
  ]
  name "R1(37)"
  boundingObject USE b1
}
Solid {
  translation -3.36681 17.3522 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 8.69 0.6
      }
    }
  ]
  name "R1(1)"
  boundingObject USE b1
}
Solid {
  translation 7.45591 10.0048 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 16.69 0.6
      }
    }
  ]
  name "R1(11)"
  boundingObject USE b1
}
Solid {
  translation 14.0684 4.46054 0.0487247
  rotation 1.13152e-17 2.94518e-17 1 2.7414
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 5.69 0.6
      }
    }
  ]
  name "R1(14)"
  boundingObject USE b1
}
Solid {
  translation 15.8577 -1.49772 0.0487247
  rotation 1.13152e-17 2.94518e-17 1 2.7414
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 5.69 0.6
      }
    }
  ]
  name "R1(19)"
  boundingObject USE b1
}
Solid {
  translation 17.3189 -10.412 0.0487247
  rotation 1.13152e-17 2.94518e-17 1 2.7414
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 3.69 0.6
      }
    }
  ]
  name "R1(45)"
  boundingObject USE b1
}
Solid {
  translation -9.99232 -1.89895 0.0487247
  rotation 1.13152e-17 2.94518e-17 1 2.7414
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 2.66 0.6
      }
    }
  ]
  name "R1(47)"
  boundingObject USE b1
}
Solid {
  translation 17.7253 2.91389 0.0487247
  rotation 1.13152e-17 2.94518e-17 1 2.7414
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 10.69 0.6
      }
    }
  ]
  name "R1(16)"
  boundingObject USE b1
}
Solid {
  translation 14.6522 9.89284 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 11.69 0.6
      }
    }
  ]
  name "R1(13)"
  boundingObject USE b1
}
Solid {
  translation 10.1177 0.485546 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 12.69 0.6
      }
    }
  ]
  name "R1(15)"
  boundingObject USE b1
}
Solid {
  translation 8.15886 -0.987752 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 14.69 0.6
      }
    }
  ]
  name "R1(17)"
  boundingObject USE b1
}
Solid {
  translation 6.31497 -3.98632 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 25.69 0.6
      }
    }
  ]
  name "R1(44)"
  boundingObject USE b1
}
Solid {
  translation 3.57867 -6.27081 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 29 0.6
      }
    }
  ]
  name "R1(46)"
  boundingObject USE b1
}
Solid {
  translation 2.56432 -8.36078 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 28.5 0.6
      }
    }
  ]
  name "R1(48)"
  boundingObject USE b1
}
Solid {
  translation 1.30586 -11.3357 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 28.5 0.6
      }
    }
  ]
  name "R1(49)"
  boundingObject USE b1
}
Solid {
  translation -3.46099 -9.30747 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 28.5 0.6
      }
    }
  ]
  name "R1(50)"
  boundingObject USE b1
}
Solid {
  translation 5.48054 5.33533 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 16.69 0.6
      }
    }
  ]
  name "R1(12)"
  boundingObject USE b1
}
Solid {
  translation -17.0755 5.00783 0.0487247
  rotation -4.61246e-17 -7.2918e-18 -1 1.44739
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 2.7 0.6
      }
    }
  ]
  name "R1(23)"
  boundingObject USE b1
}
Solid {
  translation -6.59689 7.45612 0.0487247
  rotation -4.61246e-17 -7.2918e-18 -1 1.44739
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 3 0.6
      }
    }
  ]
  name "R1(39)"
  boundingObject USE b1
}
Solid {
  translation -5.95521 5.44983 0.0487247
  rotation -4.61246e-17 -7.2918e-18 -1 1.44739
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 5 0.6
      }
    }
  ]
  name "R1(40)"
  boundingObject USE b1
}
Solid {
  translation -1.87716 1.41101 0.0487247
  rotation -4.61246e-17 -7.2918e-18 -1 1.44739
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.6 7 0.6
      }
    }
  ]
  name "R1(43)"
  boundingObject USE b1
}
Solid {
  translation -18.0176 1.8864 0.0487247
  rotation 2.53508e-17 2.05166e-17 1 -2.494585307179586
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 3.2 0.6
      }
    }
  ]
  name "R1(31)"
  boundingObject USE b1
}
Solid {
  translation -15.223 3.23221 0.0487247
  rotation -4.61246e-17 -7.2918e-18 -1 1.44739
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 2.9 0.6
      }
    }
  ]
  name "R1(25)"
  boundingObject USE b1
}
Solid {
  translation -15.7477 1.98432 0.0487247
  rotation 2.53508e-17 2.05166e-17 1 -2.494585307179586
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 2.9 0.6
      }
    }
  ]
  name "R1(26)"
  boundingObject USE b1
}
Solid {
  translation -16.7763 3.15665 0.0487247
  rotation 2.16922e-17 2.28457e-17 1 -2.756385307179586
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.3 0.3 0.6
      }
    }
  ]
  name "R1(27)"
  boundingObject USE b1
}
Solid {
  translation -18.8065 7.59751 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 5.28 0.6 0.6
      }
    }
  ]
  name "R1(21)"
  boundingObject USE b1
}
Solid {
  translation -21.4245 1.46212 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 5.55 0.6 0.6
      }
    }
  ]
  name "R1(33)"
  boundingObject USE b1
}
Solid {
  translation -15.2126 6.8041 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 3.95 0.6 0.6
      }
    }
  ]
  name "R1(22)"
  boundingObject USE b1
}
Solid {
  translation -19.1212 5.18896 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 1.9 0.6
      }
    }
  ]
  name "R1(20)"
  boundingObject USE b1
}
Solid {
  translation -9.02428 7.70382 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 2.35 0.6
      }
    }
  ]
  name "R1(38)"
  boundingObject USE b1
}
Solid {
  translation -10.3076 5.97769 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 4.4 0.6
      }
    }
  ]
  name "R1(41)"
  boundingObject USE b1
}
Solid {
  translation -19.7331 3.43844 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 1.9 0.6
      }
    }
  ]
  name "R1(32)"
  boundingObject USE b1
}
Solid {
  translation -6.19532 -24.982 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 1
        }
      }
      geometry Box {
        size 0.05 0.05 0.05
      }
    }
  ]
  name "BLUE BOX"
  boundingObject USE b1
  recognitionColors [
    0 0 1
  ]
}
Solid {
  translation -17.6531 -3.29456 0.0487247
  rotation -3.389314249066104e-17 -1.5078499169417248e-17 -1 1.9709910307427163
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 2.3 0.6
      }
    }
  ]
  name "R1(30)"
  boundingObject USE b1
}
Solid {
  translation -15.4888 11.0148 0.0509996
  rotation 0.0008590160950429561 0.0005637474055813995 -0.9999994721399663 1.9732688516300343
  children [
    DEF b1 Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.6 8.8 0.6
      }
    }
  ]
  name "R2(2)"
  boundingObject USE b1
}
